test/unit/uk/gov/hmrc/bindingtariffrulingfrontend/model/PagedTest.scala

```diff
@@ -26,28 +28,130 @@ class PagedTest extends BaseSpec {
     }
 
     "calculate size" in {
-      Paged.empty.size    shouldBe 0
-      Paged(Seq("")).size shouldBe 1
+      Paged.empty.size                     shouldBe 0
+      Paged.empty(SimplePagination()).size shouldBe 0
+      Paged(Seq("")).size                  shouldBe 1
     }
 
     "calculate isEmpty" in {
-      Paged.empty.isEmpty    shouldBe true
-      Paged(Seq("")).isEmpty shouldBe false
+      Paged.empty.isEmpty                     shouldBe true
+      Paged.empty(SimplePagination()).isEmpty shouldBe true
+      Paged(Seq("")).isEmpty                  shouldBe false
     }
 
+    // scalastyle:off magic.number
     "calculate pageCount" in {
       Paged.empty.pageCount shouldBe 0
-      Paged(results = Seq(), pageIndex = 1, pageSize = 1, resultCount  = 1).pageCount   shouldBe 1
-      Paged(results = Seq(), pageIndex = 1, pageSize = 1, resultCount  = 2).pageCount   shouldBe 2
-      Paged(results = Seq(), pageIndex = 1, pageSize = 2, resultCount  = 3).pageCount   shouldBe 2
-      Paged(results = Seq(), pageIndex = 1, pageSize = 10, resultCount = 100).pageCount shouldBe 10
-      Paged(results = Seq(), pageIndex = 1, pageSize = 1, resultCount  = 100).pageCount shouldBe 100
+      Paged(results = Seq(), pageIndex   = 1, pageSize = 1, resultCount = 1).pageCount shouldBe 1
+      Paged(results = Seq(), pageIndex   = 1, pageSize = 1, resultCount = 2).pageCount shouldBe 2
+      Paged(results = Seq(), pageIndex   = 1, pageSize = 2, resultCount = 3).pageCount shouldBe 2
+      Paged(results = Seq(), pageIndex   = 1, pageSize = 10, resultCount = 100).pageCount shouldBe 10
+      Paged(results = Seq(), pageIndex   = 1, pageSize = 1, resultCount = 100).pageCount shouldBe 100
+      Paged(results = Seq(), resultCount = 1).pageCount shouldBe 1
+      Paged(results = Seq(), resultCount = 100).pageCount shouldBe 1
+      Paged(results = Seq(), resultCount = 1000).pageCount shouldBe 10
     }
+    // scalastyle:on magic.number
 
     "calculate nonEmpty" in {
-      Paged.empty.nonEmpty    shouldBe false
-      Paged(Seq("")).nonEmpty shouldBe true
+      Paged.empty.nonEmpty                     shouldBe false
+      Paged.empty(SimplePagination()).nonEmpty shouldBe false
+      Paged(Seq("")).nonEmpty                  shouldBe true
     }
+
+    "valid results when results are provided" in {
+      val validPaged = Json
+        .obj(
+          "results"     -> Seq(""),
+          "pageIndex"   -> 1,
+          "pageSize"    -> 1,
+          "resultCount" -> 1
+        )
+        .as[Paged[String]]
+
+      validPaged shouldBe Paged(Seq(""), 1, 1, 1)
+    }
+
+    "invalid results when no results provided" in {
```

Comment:
same here the test is checking something else, more like invalid json parsing throws an exception or something similar based on the logic